gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -165,8 +165,9 @@ public void run() {
     } else {
       // Since we are going to change status to isActive=false, unschedule all flows
       for (Spec spec : this.scheduledFlowSpecs.values()) {
-        onDeleteSpec(spec.getUri(), spec.getVersion());
+        onDeleteSpec(spec.getUri(), spec.getVersion(), new Properties(), false);
```

Comment:
Can we clone the values list instead? Then we will iterate on a copy of it that would not change when we remove entries from the map.

Overload with "removeFromMap" looks a bit strange, and future maintainers will have to spend some brain cycles to understand why it was added.

Reply:
Yeah that's true, that is much simpler. I was thinking that it may be expensive to clone all the specs, but we actually only need to clone the list, and it is okay if they are referencing the same Spec objects since we are not modifying those.